samples/snap-geometry-edits-with-utility-network-rules/src/main/java/com/esri/arcgismaps/sample/snapgeometryeditswithutilitynetworkrules/components/SnapGeometryEditsWithUtilityNetworkRulesViewModel.kt

```diff
@@ -0,0 +1,514 @@
+        // Stop the geometry editor and get the updated geometry
+        val finalGeometry = geometryEditor.stop()
+
+        // Update the feature with the new geometry
+        selectedFeature?.let { feature ->
+            feature.geometry = finalGeometry
+
+            viewModelScope.launch {
+                (feature.featureTable as GeodatabaseFeatureTable).updateFeature(feature)
+                    .onFailure { error ->
+                        messageDialogVM.showMessageDialog(
+                            "Error updating feature",
+                            error.message.toString()
+                        )
+                    }
+            }
+        }
+
+        // Reset the selection
+        resetSelections()
+    }
+
+    /**
+     * Identifies the tapped screen coordinate in the provided [singleTapConfirmedEvent] and gets
+     * the asset at that location.
+     */
+    fun identify(singleTapConfirmedEvent: SingleTapConfirmedEvent) {
+
+        if (geometryEditor.isStarted.value || arcGISMap.operationalLayers.isEmpty()) {
+            return
+        }
+        viewModelScope.launch {
+            mapViewProxy.identifyLayers(
+                screenCoordinate = singleTapConfirmedEvent.screenCoordinate,
+                tolerance = 12.dp,
+                returnPopupsOnly = false,
+                maximumResults = 1
+            ).onSuccess { identifyResultList ->
+                // As we are using subtype feature layers in this sample the returned features are
+                // contained in the sublayer results
+                val identifiedFeature = identifyResultList.firstOrNull()?.sublayerResults?.firstOrNull()?.geoElements?.firstOrNull()
+                if (identifiedFeature !is ArcGISFeature) {
+                    resetSelections()
+                    return@launch
+                }
+
+                // In this sample we only allow selection of point features. If the identified
+                // feature is null or the feature is not a point feature then reset and return.
+                if (identifiedFeature.featureTable?.geometryType != GeometryType.Point) {
+                    resetSelections()
+                    return@launch
+                } else if (
+                    selectedFeature != null
+                    && identifiedFeature != selectedFeature
+                    && selectedFeature?.featureTable?.layer is FeatureLayer
+                ) {
+                    // If a feature is already selected and the tapped feature is not the selected
+                    // feature then clear the previous selection
+                    (selectedFeature?.featureTable?.layer as FeatureLayer).clearSelection()
+                }
+
+                // Update the selected feature and select it on the layer
+                selectedFeature = identifiedFeature
+                selectedFeature?.let { feature ->
+                    (feature.featureTable?.layer as FeatureLayer).selectFeature(feature)
```

Comment:
Use `as?` to keep the sample null safe: 
```suggestion
                    (feature.featureTable?.layer as? FeatureLayer)?.selectFeature(feature)
```